x/crosschain/keeper/grpc_query_cctx_rate_limit.go

```diff
@@ -224,21 +228,35 @@ func convertCctxValue(
 		// skip CoinType_Cmd
 		return sdk.NewDec(0)
 	}
-
 	// should not happen, return 0 to skip if it happens
-	if rate.LTE(sdk.NewDec(0)) {
+	if rate.IsNil() || rate.LTE(sdk.NewDec(0)) {
+		return sdk.NewDec(0)
+	}
+
+	// get foreign coin decimals
+	_, found := foreignCoinMap[chainID]
+	if !found {
+		// skip if no coin found for this chainID
+		return sdk.NewDec(0)
+	}
+	fCoin, found := foreignCoinMap[chainID][strings.ToLower(cctx.InboundTxParams.Asset)]
```

Comment:
```suggestion
	_, found := foreignCoinMap[chainID]
	if !found {
		// skip if no coin found for this chainID
		return sdk.NewDec(0)
	}
	foreignCoin, found := foreignCoinMap[chainID][strings.ToLower(cctx.InboundTxParams.Asset)]
```
Let's kep var names explicit whenever possible

Reply:
https://github.com/zeta-chain/node/pull/2060/commits/e5e4e28a8d27e5b48c808d978ed11f25b98d2ea0